File: app/src/main/java/net/daverix/ajvm/operation/LStoreOperation.java
package net.daverix.ajvm.operation;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class LStoreOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader,
                        int indexOfBytecode,
                        Frame currentFrame) throws IOException {
        int index = reader.readUnsignedByte();
        Object localVariable = currentFrame.pop();

        if(!(localVariable instanceof Integer))
            throw new IllegalStateException("variable " + localVariable + " is not an Integer");

        currentFrame.setLocalVariable(index, localVariable);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/LStoreOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class LStoreOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader,
                         indexOfBytecode: Int,
                         currentFrame: Frame) {
        val index = reader.readUnsignedByte()
        val localVariable = currentFrame.pop()

        if (localVariable !is Int)
            throw IllegalStateException("variable $localVariable is not an Integer")

        currentFrame.setLocalVariable(index, localVariable)
    }
}
